olive/systems/python_environment/python_environment_system.py

```diff
@@ -81,8 +70,39 @@ def __init__(
 
         # path to inference script
         self.inference_path = Path(__file__).parent.resolve() / "inference_runner.py"
-        self.pass_path = Path(__file__).parent.resolve() / "pass_runner.py"
-        self.device = self.accelerators[0] if self.accelerators else Device.CPU
+        self.pass_runner_path = Path(__file__).parent.resolve() / "pass_runner.py"
+        self.evaluation_runner_path = Path(__file__).parent.resolve() / "evaluation_runner.py"
+
+    def _run_command(self, script_path: Path, config_jsons: Dict[str, Any], **kwargs) -> Dict[str, Any]:
+        """Run a script with the given config jsons and return the output json."""
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_dir_path = Path(tmp_dir).resolve()
+
+            # write config jsons to files
+            args = []
+            for key, config_json in config_jsons.items():
+                config_json_path = tmp_dir_path / f"{key}.json"
+                with config_json_path.open("w") as f:
+                    json.dump(config_json, f, indent=4)
+                args.append(f"--{key} {config_json_path}")
+
+            # command to run
+            command = f"python {script_path} {' '.join(args)}"
+            # add extra args
+            for key, value in kwargs.items():
+                if value is None:
+                    continue
+                command += f" --{key} {value}"
+            # output path
+            output_path = tmp_dir_path / "output.json"
+            command += f" --output_path {output_path}"
+
+            run_subprocess(command, env=self.environ, check=True)
```

Comment:
>command [](http://example.com/codeflow?start=27&length=7)

I suppose the command will be splitted into array of string. right?

Reply:
yes the command split is handled by `run_subprocess` https://github.com/microsoft/Olive/blob/main/olive/common/utils.py#L24. It uses `shlex.split` which handles windows or posix so we can just provide the command string.